emuu-client/src/components/UserProfile/Profile.js

```diff
@@ -82,10 +92,152 @@ function Profile() {
     setProfilePic(docSnap.data().ProfilePictureUrl);
   });
 
+  const firebaseData = JSON.parse(localStorage.getItem("firebase-data"));
+
+  const autocomplete = useMemo(
+    () =>
+      createAutocomplete({
+        onStateChange({ state }) {
+          setAutocompleteState(state);
+          setSearchInput(state.query);
+          if (count === 0) {
+            setCount((count) => count + 1);
+          }
+        },
+        getSources() {
+          return [
+            {
+              sourceId: "pages-source",
+              getItemInputValue({ item }) {
+                // search item
+                return item.query;
+              },
+              getItems({ query }) {
+                // takes your search input and checks if anything that matches it exists in your dataset
+                if (!query) {
+                  return firebaseData;
+                }
+                return firebaseData.filter(
+                  (item) =>
+                    item.VideoTitle?.toLowerCase().includes(
+                      query.toLowerCase()
+                    ) ||
+                    item.Username?.toLowerCase().includes(
+                      query.toLocaleLowerCase()
+                    )
+                );
+              },
+              templates: {
+                item({ item }) {
+                  return item.VideoTitle || item.Username;
+                },
+              },
+            },
+          ];
+        },
+      }),
+    [count]
+  );
+
+  const dataSet = autocompleteState?.collections?.[0]?.items;
+  const searchResultsVideosArr = dataSet?.filter(
+    (obj) => obj.hasOwnProperty("VideoUrl") && obj.hasOwnProperty("Username")
+  );
+  const searchResultsUsersArr = dataSet?.filter(
+    (obj) => !obj.hasOwnProperty("VideoUrl") && obj.hasOwnProperty("Username")
+  );
+  const showSearchResults =
+    searchResultsVideosArr?.length > 0 || searchResultsUsersArr?.length > 0;
+
+  const userName = localStorage.getItem("displayName");
+
+  const usersArr = firebaseData.filter(
+    (obj) => obj.hasOwnProperty("Username") && !obj.hasOwnProperty("VideoUrl")
+  );
+  const videosArr = firebaseData.filter(
+    (obj) => obj.hasOwnProperty("Username") && obj.hasOwnProperty("VideoUrl")
+  );
+
+  const handleCreatorProfile = (creatorsName) => {
+    const creatorsData = usersArr.filter(
+      (user) => user.Username === creatorsName
+    );
+    const creatorsDataVideos = videosArr.filter(
+      (video) => video.Username === creatorsName
+    );
+    localStorage.setItem("creatorsData", JSON.stringify(creatorsData));
+    localStorage.setItem(
+      "creatorsDataVideos",
+      JSON.stringify(creatorsDataVideos)
+    );
+
+    {
+      location.pathname === "/creator"
+        ? window.location.reload()
+        : history.push("/creator");
+    }
+  };
```

Comment:
Again, this search feature should be its own separate function that is being called. No need to write the same code on every page.